src/editor.js

```diff
@@ -41,6 +41,51 @@ import htmlMinifier from 'html-minifier';
 
 const {id, g, n, s} = getNamespace('editor');
 
+/**
+ * Holds CodeMirror HTML autocompletion specs for AMP formats.
+ * This is taken from amp.dev and is built from the AMP spec... somehow.
+ */
+const hintsUrl = '/static/hints/amphtml-hint.json';
+
+/**
+ * Hints ignored on typing when delimited by these chars.
+ * Stolen from @ampproject/docs/playground/src/editor/editor.js
+ */
+const hintIgnoreEnds = new Set([
+  ';',
+  ',',
+  ')',
+  '`',
+  '"',
+  "'",
+  '>',
+  '{',
+  '}',
+  '[',
+  ']',
+]);
+
+/**
+ * tagName-to-attributes for uploaded file autocomplete hints.
+ * (These only apply to the amp4ads validation set.)
+ *
+ * TODO(alanorozco): build this set dynamically
+ * TODO(alanorozco): set hints for tag context based on file extension
+ */
+const attrFileHintTagAttrs = {
```

Comment:
Lets remove extensions that are not allowed in a4a [whitelist](https://amp.dev/documentation/guides-and-tutorials/learn/a4a_spec#allowed-amp-extensions-and-builtins)

Reply:
Ah, you're right, I did not notice that. Interesting, I took these from the hint set. There might be something wrong with how the hint set is generated, we might need to look into that later.